engine/src/witness/arb.rs

```diff
@@ -0,0 +1,175 @@
+				state_chain_client.latest_finalized_block().hash,
+			)
+			.await
+			.context("Failed to fetch Arbitrum supported assets")?;
+
+	let usdc_contract_address = *supported_arb_erc20_assets
+		.get(&cf_primitives::chains::assets::arb::Asset::ArbUsdc)
+		.context("ArbitrumSupportedAssets does not include USDC")?;
+
+	let supported_arb_erc20_assets: HashMap<H160, cf_primitives::Asset> =
+		supported_arb_erc20_assets
+			.into_iter()
+			.map(|(asset, address)| (address, asset.into()))
+			.collect();
+
+	let arb_source = EvmSource::<_, Arbitrum>::new(arb_client.clone())
+		.strictly_monotonic()
+		.shared(scope);
+
+	arb_source
+		.clone()
+		.chunk_by_time(epoch_source.clone(), scope)
+		.chain_tracking(state_chain_client.clone(), arb_client.clone())
+		.logging("chain tracking")
+		.spawn(scope);
+
+	let vaults = epoch_source.vaults::<Arbitrum>().await;
+
+	// ===== Full witnessing stream =====
+
+	let arb_safety_margin = state_chain_client
+		.storage_value::<pallet_cf_ingress_egress::WitnessSafetyMargin<
+			state_chain_runtime::Runtime,
+			state_chain_runtime::ArbitrumInstance,
+		>>(state_chain_stream.cache().hash)
+		.await?
+		// Default to berghain in case the value is missing (e.g. during initial upgrade)
+		.unwrap_or(ARBITRUM_SAFETY_MARGIN);
+
+	tracing::info!("Safety margin for Arbitrum is set to {arb_safety_margin} blocks.",);
+
+	let arb_safe_vault_source = arb_source
+		.lag_safety(arb_safety_margin as usize)
+		.logging("safe block produced")
+		.chunk_by_vault(vaults, scope);
+
+	let arb_safe_vault_source_deposit_addresses = arb_safe_vault_source
+		.clone()
+		.deposit_addresses(scope, state_chain_stream.clone(), state_chain_client.clone())
+		.await;
+
+	arb_safe_vault_source
+		.clone()
+		.key_manager_witnessing(process_call.clone(), arb_client.clone(), key_manager_address)
+		.continuous("ArbitrumKeyManager".to_string(), db.clone())
+		.logging("ArbitrumKeyManager")
+		.spawn(scope);
+
+	arb_safe_vault_source_deposit_addresses
+		.clone()
+		.erc20_deposits::<_, _, _, UsdcEvents>(
+			process_call.clone(),
+			arb_client.clone(),
+			cf_primitives::chains::assets::arb::Asset::ArbUsdc,
+			usdc_contract_address,
+		)
+		.await?
+		.continuous("ArbitrumUSDCDeposits".to_string(), db.clone())
+		.logging("ArbitrumUSDCDeposits")
+		.spawn(scope);
+
+	arb_safe_vault_source_deposit_addresses
+		.clone()
+		.ethereum_deposits(
+			process_call.clone(),
+			arb_client.clone(),
+			cf_primitives::chains::assets::arb::Asset::ArbEth,
+			address_checker_address,
+			vault_address,
+		)
+		.await
+		.continuous("ArbitrumDeposits".to_string(), db.clone())
+		.logging("ArbitrumDeposits")
+		.spawn(scope);
+
+	arb_safe_vault_source
+		.vault_witnessing(
+			process_call,
+			arb_client.clone(),
+			vault_address,
+			cf_primitives::Asset::ArbEth,
+			cf_primitives::ForeignChain::Arbitrum,
+			supported_arb_erc20_assets,
+		)
+		.continuous("ArbitrumVault".to_string(), db)
+		.logging("ArbitrumVault")
```

Comment:
no need to add `Arbitrum` on the logging, the chain is already included for the logging adapter

Reply:
same for all the above